java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CuVSHostMatrixImpl.java

```diff
@@ -131,6 +130,12 @@ public ValueLayout valueLayout() {
     return valueLayout;
   }
 
+  @Override
+  public MemorySegment toTensor(Arena arena) {
```

Comment:
This is a clean way to do it:  Host matrices having a method to convert to the appropriate tensor.